### ISSUE TITLE
Actions: Add API endpoint for single-row `UPDATE` (first pass)

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#22544]
Part of #22542
Follow-on to #22543. 

This only needs to be for Postgres for now but in the future we want to support other drivers so it would probably make sense to add a new driver method for doing this so we can add support for more drivers in the future. The endpoint should take a Table ID, primary key value(s), and a map of column name -> value and update the values in the row `PATCH` style (only update the values that were passed in). 

Worth remembering that a table might have a compound primary key (multiple columns) or weird types as the primary key; we shouldn't just assume everything uses an integer ID or something like that.

For the first pass we don't need to do a lot of validation e.g. checking that the value types match column types; we can let the database worry about that and report the errors in the API response. It's probably still worth passing in the values with a `::type` conversion so Postgres can do the right thing and cast say a string to a `timestamp with time zone` if that's how we pass it in.

```sql
UPDATE my_table
SET some_timestamp = ?::timestamptz
WHERE id = ?
```

This can be synchronous for the time being.

**Comment by camsaul** [comment]
- Once #22551 is implemented, we should disable this endpoint if the feature flag is disabled. It's probably ok to do that for the `/api/actions` prefix as a whole. I've opened issue #22555 to track disabling endpoints when the flag is disabled.
- Once #22553 is implemented, we should disable this endpoint if actions are disabled for the affected Database. I've opened issue #22556 to track disabling endpoints when actions are disabled for a particular Database
- Once #22557 is implemented we may or may not want to disable this endpoint if the driver does not support actions (TBD)
